tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -3346,3 +3346,146 @@ def test_13_cleanup(self):
+
+            # create a workload
+            workload_id = self.workload_create([vm_id],
+                                               tvaultconf.workload_type_id, encryption=True,
+                                               secret_uuid=secret_uuid)
+
+            LOG.debug("Workload ID: " + str(workload_id))
+            reporting.add_test_step("Encrypted Workload with encrypted and non encrypted volume is created.",
+                                    tvaultconf.PASS)
+
+            # Create full snapshot
+            self.snapshot_id_full = self.workload_snapshot(
+                workload_id, True)
+            self.wait_for_workload_tobe_available(workload_id)
+            if (self.getSnapshotStatus(workload_id, self.snapshot_id_full) != "available"):
+                self.exception = "Create full snapshot"
+                raise Exception(str(self.exception))
+
+            self.snapshot_ids.append(self.snapshot_id_full)
+
+            LOG.debug("Snapshot ID-1: " + str(self.snapshot_ids[0]))
+            reporting.add_test_step("Full Snapshot is created.",
+                                    tvaultconf.PASS)
+
+
+            # Create incremental-1 snapshot
+            self.snapshot_id_inc = self.workload_snapshot(
+                workload_id, False)
+            self.wait_for_workload_tobe_available(workload_id)
+            if (self.getSnapshotStatus(workload_id, self.snapshot_id_inc) != "available"):
+                self.exception = "Create incremental-1 snapshot"
+                raise Exception(str(self.exception))
+
+            self.snapshot_ids.append(self.snapshot_id_inc)
+            LOG.debug("Snapshot ID-2: " + str(self.snapshot_ids[1]))
+            reporting.add_test_step("Incremental Snapshot is created.",
+                                    tvaultconf.PASS)
+
+            volume_snapshots1 = self.get_volume_snapshots(volume_id)
+            LOG.debug("Volumes snapshots for: " +
+                      str(volume_id) + ": " + str(volume_snapshots1))
+            if len(volume_snapshots1) == 1:
+                reporting.add_test_step("Unencrypted Volume snapshot count is 1", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Unencrypted Volume snapshot count is not 1", tvaultconf.FAIL)
+                raise Exception("Unencrypted Volume snapshot count is not 1")
+
+            volume_snapshots2 = self.get_volume_snapshots(encryp_volume_id)
+            LOG.debug("Volumes snapshots for: " +
+                      str(encryp_volume_id) + ": " + str(volume_snapshots2))
+            if len(volume_snapshots2) == 1:
+                reporting.add_test_step("Encrypted Volume snapshot count is 1", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Encrypted Volume snapshot count is not 1", tvaultconf.FAIL)
+                raise Exception("Encrypted Volume snapshot count is not 1")
+
+            self.workload_reset(workload_id)
```

Comment:
Add validation for this call and result to report accordingly

Reply:
done